src/features/defi/providers/cosmos/components/CosmosManager/Claim/ClaimStatus.tsx

```diff
@@ -0,0 +1,160 @@
+import { Box, Button, Center, Link, ModalBody, ModalFooter, Stack } from '@chakra-ui/react'
+import { AssetId, ChainId } from '@shapeshiftoss/caip'
+import { useEffect, useState } from 'react'
+import { FaCheck, FaTimes } from 'react-icons/fa'
+import { useTranslate } from 'react-polyglot'
+import { useLocation } from 'react-router'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import { IconCircle } from 'components/IconCircle'
+import { MiddleEllipsis } from 'components/MiddleEllipsis/MiddleEllipsis'
+import { Row } from 'components/Row/Row'
+import { SlideTransition } from 'components/SlideTransition'
+import { RawText } from 'components/Text'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+interface ClaimStatusState {
+  txid: string
+  assetId: AssetId
+  amount: string
+  userAddress: string
+  estimatedGas: string
+  usedGasFee?: string
+  status: string
+  chainId: ChainId
+}
+
+enum TxStatus {
+  PENDING = 'pending',
+  SUCCESS = 'success',
+  FAILED = 'failed',
+}
+
+type ClaimState = {
+  txStatus: TxStatus
+  usedGasFee?: string
+}
+
+const StatusInfo = {
+  [TxStatus.PENDING]: {
+    text: 'defi.broadcastingTransaction',
+    color: 'blue.500',
+  },
+  [TxStatus.SUCCESS]: {
+    text: 'defi.transactionComplete',
+    color: 'green.500',
+    icon: <FaCheck />,
+  },
+  [TxStatus.FAILED]: {
+    text: 'defi.transactionFailed',
+    color: 'red.500',
+    icon: <FaTimes />,
+  },
+}
+
+export const ClaimStatus = () => {
+  const { history: browserHistory } = useBrowserRouter()
+  const translate = useTranslate()
+  const {
+    state: { txid, amount, assetId, userAddress },
+  } = useLocation<ClaimStatusState>()
+  const [state, setState] = useState<ClaimState>({
+    txStatus: TxStatus.PENDING,
+  })
+
+  // Asset Info
+  const asset = useAppSelector(state => selectAssetById(state, assetId)) // TODO: diff denom for rewards
+
+  useEffect(() => {
+    ;(async () => {
+      if (!txid) return
+      try {
+        setState({
+          ...state,
+          txStatus: txid ? TxStatus.SUCCESS : TxStatus.FAILED,
+        })
+      } catch (error) {
+        console.error('CosmosClaim:useEffect error:', error)
```

Comment:
moduleLogger?

Reply:
These are all copied from the original implementation, if that's fine by you I'll get started with a PR stacked on top of this one (the very parent one) that tackles all console.log / console.errors both on the new implementation as well as the Yearn/Foxy ones?
This way we'll have a clean similar diff across all three

Reply:
https://github.com/shapeshift/web/pull/2323